ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx

```diff
@@ -0,0 +1,219 @@
+  useRef,
+  useEffect,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+import { I18nContext } from '../../../../../contexts/i18n';
+///: END:ONLY_INCLUDE_IF
+import Confusable from '../../../../ui/confusable';
+import {
+  AvatarAccount,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  AvatarIcon,
+  AvatarIconSize,
+  BadgeWrapper,
+  IconName,
+  ///: END:ONLY_INCLUDE_IF
+  Text,
+} from '../../../../component-library';
+import {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  BackgroundColor,
+  BorderColor,
+  IconColor,
+  ///: END:ONLY_INCLUDE_IF
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { ellipsify } from '../../../../../pages/confirmations/send/send.utils';
+import Tooltip from '../../../../ui/tooltip';
+
+type DomainInputResolutionCellArgs = {
+  domainType: string;
+  address: string;
+  protocol?: string;
+  domainName: string;
+  resolvingSnap?: string;
+  onClick: () => void;
+};
+
+export const DomainInputResolutionCell = ({
+  domainType,
+  address,
+  domainName,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  resolvingSnap = '',
+  ///: END:ONLY_INCLUDE_IF
+  onClick,
+  protocol,
+}: DomainInputResolutionCellArgs) => {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  const t: (key: string, params: unknown[]) => string = useContext(I18nContext);
+  ///: END:ONLY_INCLUDE_IF
+  const titleRef = useRef<null | HTMLDivElement>(null);
+  const breakpointRef = useRef<null | number>(null);
+  const [isTitleOverflowing, setIsTitleOverflowing] = useState(false);
+
+  useEffect(() => {
+    if (!titleRef.current) {
+      return;
+    }
+
+    let isOverflowing =
+      titleRef.current.offsetWidth < titleRef.current.scrollWidth;
+    const breakpointLength = titleRef.current.textContent?.length;
+
+    if (isOverflowing && !breakpointRef.current && breakpointLength) {
+      breakpointRef.current = breakpointLength;
+    }
+
+    if (!isOverflowing) {
+      if (breakpointRef.current) {
+        if (domainName.length >= breakpointRef.current) {
+          isOverflowing = true;
+        } else {
+          isOverflowing = false;
+          breakpointRef.current = null;
+        }
+      }
+    }
+
+    if (isOverflowing !== isTitleOverflowing) {
+      setIsTitleOverflowing(isOverflowing);
+    }
+  }, [domainName, isTitleOverflowing]);
+
+  const OverflowingTitle = () => (
+    <Tooltip
+      containerClassName="multichain-send-page__recipient__item__title-tooltip"
+      wrapperClassName="multichain-send-page__recipient__item__title-tooltip-container"
+      position="bottom"
+      title={domainName}
+    >
+      <Confusable
+        asText
+        input={domainName}
+        confusableWrapperName="multichain-send-page__recipient__item__title-confusable-wrapper"
+      />
+    </Tooltip>
+  );
+
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  if (domainType === 'Other') {
+    // Snap provided resolution.
+    return (
+      <div
```

Comment:
Instead of using `<div>`, can we use `<Box>` instead?  That's been our practice for a year or so now.

Reply:
Also, could the be separated into its own component?  Seems a decent sized alternative block.

Reply:
I'd like to keep as is, I think it would increase complexity to make a distinction between `ENS` and snap related resolution components, especially when the ENS resolution is being planned to be rolled into a snap in the future.